Code/GraphMol/MolOps.cpp

```diff
@@ -383,6 +383,48 @@ void sanitizeMol(RWMol &mol, unsigned int &operationThatFailed,
   operationThatFailed = 0;
 }
 
+std::vector<std::unique_ptr<MolSanitizeException>> detectChemistryProblems(
+    const ROMol &imol, unsigned int sanitizeOps) {
+  RWMol mol(imol);
+  std::vector<std::unique_ptr<MolSanitizeException>> res;
+
+  // clear out any cached properties
+  mol.clearComputedProps();
+
+  int operation;
+  operation = SANITIZE_CLEANUP;
+  if (sanitizeOps & operation) {
+    // clean up things like nitro groups
+    cleanUp(mol);
+  }
+
+  // update computed properties on atoms and bonds:
+  operation = SANITIZE_PROPERTIES;
+  if (sanitizeOps & operation) {
+    for (auto &atom : mol.atoms()) {
+      try {
+        bool strict = true;
```

Comment:
agreed that it could, and normally should.
For these cases where I'm basically using the variable name to document the meaning of the optional argument I like to have the variable declarations close to the point where they're used.